.github/CONTRIBUTING.md

```diff
@@ -29,16 +19,61 @@ Facebook has a [bounty program](https://www.facebook.com/whitehat/) for the safe
 disclosure of security bugs. In those cases, please go through the process
 outlined on that page and do not file a public issue.
 
+## Pull Requests
+
+### Guidelines
+
+We actively welcome your pull requests.
+
+However, if you're adding anything significant (e.g. > 50 lines), please
+open up an issue before sending a PR to involve and discuss with maintainers if the change is required and will be accepted.
+
+We take the following factors into consideration before accepting features and PRs:
+
+1. If the feature can be achieved without modifying MMF. MMF is designed so that you can implement extensions from the outside, e.g. [Hateful Memes ConcatVL example](https://github.com/apsdehal/hm_example_mmf).
+  * If something is not extensible, please open up an issue so we can improve MMF together.
+1. Whether the feature is potentially useful to a large audience (e.g. an impactful paper, a popular dataset, a significant speedup or a widely useful utility).
+1. Whether the proposed solution has a good design/interface. This can be discussed in the issues prior to the PRs, or in the form of a draft PR and we will help you make it better.
+1. Whether the proposed solution adds extra overhead to users who don't need such feature.
+1. Whether the proposed solution breaks existing APIs.
+
+### Process
+
+1. Read the [PR guidelines](#guidelines) if you haven't.
+1. Fork the repo and create your branch from `master`.
+1. If your PR contains multiple orthogonal changes, split it to several PRs. Keep one PR focused on a single change while keeping it small.
+1. If you've added code that should be tested, add tests.
+1. Follow the [coding style guidelines](#coding-style) mentioned below.
+1. If you've changed APIs:
+    * Update the documentation. We use the [Google style docstrings](https://www.sphinx-doc.org/en/master/usage/extensions/napoleon.html) in python.
+    * Ensure backwards compatibility.
+1. Ensure the test suite passes.
+1. If the PR involves adding (i) a new dataset: update the dataset zoo to enable automatic downloads (ii) a new model: you don't need to update the model zoo, but make sure to describe your results and experiments in PR description.
```

Comment:
We should also ask here to add to our docs pages for [models](https://mmf.sh/docs/notes/model_zoo) and [datasets](https://mmf.sh/docs/notes/dataset_zoo).